packages/material-ui-styles/src/ThemeProvider/ThemeProvider.js

```diff
@@ -2,8 +2,8 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import warning from 'warning';
 import { exactProp } from '@material-ui/utils';
-import ThemeContext from './ThemeContext';
-import useTheme from './useTheme';
+import ThemeContext from '../useTheme/ThemeContext';
```

Comment:
I guess it's the most controversial change I can find in the pull request. I do prefer hosting ThemeContext in the consumer than in the provider. People might not use the provider, but will always use the consumer.

Reply:
I had to make a choice. I decided this way.

ThemeProvider was already importing useTheme
But useTheme wasn't importaing ThemeProvider
So I wanted to avoid to create something similar to a circular reference